src/components/UserSelector/UserSelector.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useCallback, useContext, useState } from 'react';
+import cn from 'classnames';
+import { getPosts } from '../../api/posts';
+import { MainContext } from '../MainContext/MainContext';
+import { User } from '../../types/User';
+import { Errors } from '../../types/Errors';
+
+type Props = {
+  choosedUser?: User,
+  setChoosedUser: (value: User) => void,
+};
+
+export const UserSelector: React.FC<Props> = ({
+  choosedUser,
+  setChoosedUser,
+}) => {
+  const {
+    users,
+    setPosts,
+    setError,
+    setIsLoading,
+    setChoosedPost,
+  } = useContext(MainContext);
+
+  const [isActive, setIsActive] = useState(false);
+
+  const handleChooseUser = (user: User) => {
+    setIsLoading(true);
+    setChoosedUser(user);
+    setIsActive(false);
+    setPosts([]);
+    setChoosedPost(null);
+    setError(Errors.NONE);
+
+    getPosts(user.id)
+      .then(setPosts)
+      .catch(() => setError(Errors.POST))
+      .finally(() => setIsLoading(false));
+  };
+
+  const handleBlur = useCallback(() => setTimeout(() => {
+    if (isActive) {
+      setIsActive(false);
+    }
+  }, 170), [isActive]);
+
+  return (
+    <div
+      data-cy="UserSelector"
+      className={cn('dropdown', {
+        'is-active': isActive,
+      })}
+      onBlur={handleBlur}
+    >
+      <div className="dropdown-trigger">
+        <button
+          type="button"
+          className="button"
+          aria-haspopup="true"
+          aria-controls="dropdown-menu"
+          onClick={() => setIsActive(!isActive)}
```

Comment:
move to an own function an use `previous` value
```suggestion
          onClick={() => setIsActive(prev => !prev)}
```